code/modules/mob/living/carbon/death.dm

```diff
@@ -1,5 +1,5 @@
 /mob/living/carbon/death(gibbed)
-	if(stat == DEAD)
+	if(stat == DEAD || HAS_TRAIT(src, TRAIT_NODEATH) && !gibbed)
```

Comment:
I don't think this should be necessary, instead you should find what is calling `death` without respecting nodeath

Reply:
In that case I'll need another PR for the death handling thing, I'll remove it and keep the removal of crit damage for them.